src/opdsrequestmanager.cpp

```diff
@@ -21,6 +26,44 @@ int OpdsRequestManager::getCatalogPort()
          : 443;
 }
 
+// Helper to handle replies and follow redirects
+void OpdsRequestManager::handleReply(QNetworkReply* reply, std::function<void(QNetworkReply*)> finalHandler, int redirectCount) {
+    QVariant redirectAttr = reply->attribute(QNetworkRequest::RedirectionTargetAttribute);
+    if (redirectAttr.isValid() && redirectCount < kMaxRedirects) {
+        QUrl redirectUrl = redirectAttr.toUrl();
+        // Make absolute if needed
+        if (redirectUrl.isRelative()) {
+            redirectUrl = reply->url().resolved(redirectUrl);
+        }
+        qInfo() << "Following redirect" << redirectCount + 1 << "to:" << redirectUrl.toString();
+        reply->deleteLater();
+        QNetworkRequest newRequest(redirectUrl);
+        QNetworkReply* newReply = m_networkManager.get(newRequest);
+        connect(newReply, &QNetworkReply::finished, this, [=]() {
+            handleReply(newReply, finalHandler, redirectCount + 1);
+        });
```

Comment:
I would extract this piece of code into a function of its own `followRedirect(reply, finalHandler, redirectCount)` (in order to keep `handleReply()` short and readable.